R/variance_ratio.R

```diff
@@ -33,100 +33,162 @@
-#' 
-#' Null model confidence intervals default to the standard lowest 2.5\% and 
-#' upper 97.5\% of the null distribution, typically these do not need to be change, 
+#'
+#' Null model confidence intervals default to the standard lowest 2.5\% and
+#' upper 97.5\% of the null distribution, typically these do not need to be change,
 #' but they can be user-modified to set more stringent CIs.
 #'  @references
-#'  Hallett, Lauren M., Joanna S. Hsu, Elsa E. Cleland, Scott L. Collins, 
-#'  Timothy L. Dickson, Emily C. Farrer, Laureano A. Gherardi, et al. (2014) 
-#'  "Biotic Mechanisms of Community Stability Shift along a Precipitation Gradient." 
+#'  Hallett, Lauren M., Joanna S. Hsu, Elsa E. Cleland, Scott L. Collins,
+#'  Timothy L. Dickson, Emily C. Farrer, Laureano A. Gherardi, et al. (2014)
+#'  "Biotic Mechanisms of Community Stability Shift along a Precipitation Gradient."
 #'  Ecology 95, no. 6: 1693-1700. doi: 10.1890/13-0895.1
-#'  
-#'  Schluter, Dolph. (1984) "A Variance Test for Detecting Species Associations, 
+#'
+#'  Schluter, Dolph. (1984) "A Variance Test for Detecting Species Associations,
 #'  with Some Example Applications." Ecology 65, no. 3: 998-1005. doi:10.2307/1938071.
 #' @export
-#' @examples 
+#' @examples
 #'  data(knz_001d)
-#'  
+#'
 #'  # Calculate the variance ratio and CIs averaged within replicates
-#'  # Here the null model is repeated once, for final use it is recommended to set a 
+#'  # Here the null model is repeated once, for final use it is recommended to set a
 #'  # large bootnumber (eg, 10000)
-#'  
-#'  res_averagedreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species", 
+#'
+#'  res_averagedreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species",
 #'  abundance.var = "abundance", bootnumber = 1, replicate = "subplot")
-#'  
+#'
 #'  #Calculate the variance ratio and CIs for each replicate
-#'  
-#'  res_withinreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species", 
+#'
+#'  res_withinreplicates <- variance_ratio(knz_001d, time.var = "year", species.var = "species",
 #'  abundance.var = "abundance", bootnumber = 1, replicate = "subplot", average.replicates = FALSE)
-variance_ratio <- function(df, time.var, species.var, abundance.var, bootnumber, replicate.var=NA,
-                        li=0.025, ui=0.975,  average.replicates=TRUE) {
-  
+variance_ratio <- function(df, time.var, species.var, abundance.var,
+                           bootnumber, replicate.var,
+                           average.replicates = TRUE, level = 0.95, ...) {
+
   # check to make sure abundance is numeric data
   check_numeric(df, time.var, abundance.var)
-   
+
   # if no replicates, calculate single variance ratio
-   if(is.na(replicate.var)) {
-        check_single_onerep(df, time.var, species.var)
-        VR <- variance_ratio_longformdata(df, time.var, species.var, abundance.var)
+  if  (missing(replicate.var)) {
```

Comment:
Okay. Don't think it matters too much as long as the default was NA. We'll
have to change this across all the functions for consistency

On Fri, Mar 18, 2016 at 1:01 AM, Andrew MacDonald <notifications@github.com>
wrote:

> In R/variance_ratio.R
> <https://github.com/laurenmh/codyn/pull/67#discussion_r56619165>:
>
> >    # if no replicates, calculate single variance ratio
> > -   if(is.na(replicate.var)) {
> > -        check_single_onerep(df, time.var, species.var)
> > -        VR <- variance_ratio_longformdata(df, time.var, species.var, abundance.var)
> > +  if  (missing(replicate.var)) {
>
> in variance_ratio I decided to test for missing, rather than NA, column
> names. it seems to me that if there is no replicate column, people would
> rather not specify it than specify it as NA
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/laurenmh/codyn/pull/67/files/f3e38639882cb3e07b489b62de4bd6214853b215#r56619165>
>